src/pages/Location/Location.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from "react";
+
+import { getLocation, getUrl } from "../../api";
+
+import Layout from "../../components/Layout";
+import CharacterCard from "../../components/CharacterCard";
+
+const makePromise = (urls = []) => {
+  return urls.map((url) => getUrl(url));
+};
+
+class Location extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      location: null,
+      residents: [],
+      hasLoaded: false,
+      hasError: false,
+      errorMessage: null,
+    };
+    this.loadLocation = this.loadLocation.bind(this);
+  }
+
+  componentDidMount() {
+    const { match } = this.props;
+    const { locationId } = match.params;
+    this.loadLocation(locationId);
+  }
+
```

Comment:
We should ad a `componentDidUpdate()` so when we click on a different location, form this page, the whole page refreshes